tests/datasource/fluent/integration/test_connections.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import random
+from typing import TYPE_CHECKING
+
+import pytest
+import sqlalchemy as sa
+from pytest import param
+
+from great_expectations.datasource.fluent import (
+    SnowflakeDatasource,
+    TestConnectionError,
+)
+
+if TYPE_CHECKING:
+    from sqlalchemy.engine.reflection import Inspector
+
+    from great_expectations.data_context import AbstractDataContext as DataContext
+
+
+@pytest.mark.snowflake
+class TestSnowflake:
+    @pytest.mark.parametrize(
+        "connection_string",
+        [
+            param(
+                "snowflake://ci:${SNOWFLAKE_CI_USER_PASSWORD}@${SNOWFLAKE_CI_ACCOUNT}/ci/public?warehouse=ci",
+                id="missing role",
+            ),
+            param(
+                "snowflake://ci:${SNOWFLAKE_CI_USER_PASSWORD}@${SNOWFLAKE_CI_ACCOUNT}?warehouse=ci&role=ci",
+                id="missing database + schema",
+            ),
+            param(
+                "snowflake://ci:${SNOWFLAKE_CI_USER_PASSWORD}@${SNOWFLAKE_CI_ACCOUNT}/ci?warehouse=ci&role=ci",
+                id="missing schema",
+            ),
+        ],
+    )
```

Comment:
@anthonyburdi I haven't actually changed the implementation of `TableAsset.test_connection()` yet so I was expecting these all to fail.
However, none of them do.

I'm using the sqlalchemy inspector to select a `table`, then I ensure that table cannot be queried (it raises a sqlalchemy programming error), but in each of the cases, the `.test_connection()` actually does fail.

Do you have any suggestions for how to reproduce the issue of the asset being un-selectable but `test_connection()` passing?

Reply:
I'm guessing it is because the `ci` user has [select](https://docs.snowflake.com/en/user-guide/security-access-control-privileges#table-privileges) permissions. How about creating another user `ci_without_select_permissions` and using that in this test? I think the specific issue we (me and @josectobar) were seeing was more about the select (or [usage](https://docs.snowflake.com/en/user-guide/security-access-control-privileges#schema-privileges)) permissions rather than the connection string parameters.